src/megatron/bridge/recipes/qwen_vl/data/energon/task_encoder.py

```diff
@@ -0,0 +1,596 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import json
+import logging
+import pickle
+import re
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List
+
+import numpy as np
+import torch
+from megatron.energon import Batch, DefaultTaskEncoder
+from megatron.energon.flavors.base_dataset import Sample
+from megatron.energon.task_encoder.cooking import Cooker, basic_sample_keys
+from PIL import Image
+
+from megatron.bridge.training.utils.visual_inputs import Qwen2_5_VLVisualInputs
+
+
+# Local replacements for former nemo dependencies
+# Constants for internal multimodal token placeholders and label ignore
+IGNORE_INDEX = -100
```

Comment:
for task encoder part, can we update to use https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/src/megatron/bridge/data/vlm_datasets/collate.py#L468 the same logic here. In nemo2 we split tokenizer and vision processor for some legacy work but it should be easier to combine into one.